png_benchmark.py

```diff
@@ -0,0 +1,225 @@
+"""
+Benchmarks PNG compression time and size (the ratio of the compressed image to the original image)
+on a given set of images (300 landing pad images captured from a flight test).
+
+Creates a folder with a compressed image for each quality setting to visually check the quality,
+as well as a .json with the test data and a .csv which provides a more human-friendly summary
+of the data.
+"""
+import gc
+import io
+import json
+import os
+import pathlib
+import time
+
+from PIL import Image
+
+
+# Setting parameters
+FRAME_COUNT = 300  # Total number of frames
+FRAME_TO_SAVE = 69  # This frame is good, it has both landing pads in it
+INPUT_PATH = pathlib.Path("test_images", "Encode Test Dataset 2024")
+OUTPUT_PATH = pathlib.Path(f"log_{int(time.time())}")
+COMPRESS_TYPES = [0, 1, 2, 3, 4]  # There are 5 different compression algorithms, each are numbered
+INITIAL_COMPRESS_LEVEL = 1  # Compress level 0 is skipped because it is uncompressed
+MAX_COMPRESS_LEVEL = 6  # Although this maxes out at 9, it takes way too long (like 10s per image)
+
+# Keys for dictionary entries
+MAX_TIME_MS = "max_time_ms"
+MIN_TIME_MS = "min_time_ms"
+AVG_TIME_MS = "avg_time_ms"
+MAX_SIZE_B = "max_size_B"
+MIN_SIZE_B = "min_size_B"
+AVG_SIZE_B = "avg_size_B"
+MAX_SIZE_RATIO_COMPRESSED_TO_ORIGINAL = "max_size_ratio_compressed_to_original_%"
+MIN_SIZE_RATIO_COMPRESSED_TO_ORIGINAL = "min_size_ratio_compressed_to_original_%"
+AVG_SIZE_RATIO_COMPRESSED_TO_ORIGINAL = "avg_size_ratio_compressed_to_original_%"
+FRAME_DATA = "frame_data"
+
+# For output csv file
+HEADERS = [
+    "Compression Type",
+    "Compression Level",
+    "Min Time (ms)",
+    "Max Time (ms)",   
+    "Avg Time (ms)",
+    "Min Size (B)",
+    "Max Size (B)",
+    "Avg Size (B)",
+    "Min Size Ratio (compressed to original in %)",
+    "Max Size Ratio (compressed to original in %)",
+    "Avg Size Ratio (compressed to original in %)",
+]
+HEADER_LINE = ",".join(HEADERS) + "\n"
+
+
+def update_min_max(min, max, current_value):
```

Comment:
Rename `min` and `max` so that they do not shadow the built in functions `min` and `max` .